include/fuzzuf/executor/traits/afl_capable.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2022 Ricerca Security
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+/**
+ * @file afl_capable.hpp
+ * @author Ricerca Security <fuzzuf-dev@ricsec.co.jp>
+ */
+
+#ifndef FUZZUF_INCLUDE_EXECUTOR_TRAITS_AFL_CAPABLE_HPP
+#define FUZZUF_INCLUDE_EXECUTOR_TRAITS_AFL_CAPABLE_HPP
+
+#include "fuzzuf/executor/native_linux_executor.hpp"
+#include "fuzzuf/feedback/exit_status_feedback.hpp"
+#include "fuzzuf/feedback/inplace_memory_feedback.hpp"
+#include "fuzzuf/utils/common.hpp"
+
+/**
+ * @class DynAFLCapable
+ * @brief Represents minimal requirements for an AFL-capable executor.
+ * 
+ * @details The executor for AFL must have the methods declared in this class.
+ * This class is to perform type erasure for the executor class to abstract
+ * the executor on the algortihms. We found that the boost's type_erasure does
+ * not meet our needs, so we implemented our own type erasure.
```

Comment:
Here, we should leave comments about the restriction imposed on `T`. 
(The dereferenced type of ) `T` should have 4 functions right? Please describe what each function should do.
Of course you can instead write the explanation of each function directly above line 55, 59, 63, 67.

Reply:
Added comments for Doxygen in [60df237](https://github.com/fuzzuf/fuzzuf/pull/41/commits/60df23722f7f1ab268855245b1df770409eadd5f) and [c202523](https://github.com/fuzzuf/fuzzuf/pull/41/commits/c20252346ef38059b0dcdc0bd55a3f595d0fafdc).